src/jack.py

```diff
@@ -1328,6 +1383,141 @@ def get_ports(self, name_pattern='', is_audio=False, is_midi=False,
             _lib.jack_free)
         return self._port_list_from_pointers(names)
 
+    def set_property(self, subject, key, value, type=''):
+        """Set a metadata property on *subject*.
+
+        Parameters
+        ----------
+        subject : int or str
+            The subject (UUID) to set the property on.
+            UUIDs can be obtained with `Client.uuid`, `Port.uuid` and
+            `Client.get_uuid_for_client_name()`.
+        key : str
+            The key (URI) of the property.  Some predefined keys are
+            available as ``jack.METADATA_*`` module constants.
+        value : str or bytes
+            The value of the property.
+        type : str, optional
+            The type of the property, either a MIME type or URI.
+            If *type* is empty, the *value* is assumed to be a UTF-8
+            encoded string (``'text/plain'``).
+
+            Example values:
+
+            - ``'image/png;base64'`` (base64 encoded PNG image)
+            - ``'http://www.w3.org/2001/XMLSchema#int'`` (integer)
+
+            Official types are preferred, but clients may use any
+            syntactically valid MIME type (which start with a type and
+            slash, like ``'text/...'``).  If a URI type is used, it must
+            be a complete absolute URI (which start with a scheme and
+            colon, like ``'http:'``).
+
+        See Also
+        --------
+        get_property
+        get_properties
+        get_all_properties
+        remove_property
+        remove_properties
+        remove_all_properties
+        set_property_change_callback
+
+        """
+        subject = _uuid_parse(subject)
+        if isinstance(value, str):
+            value = value.encode()
+        if isinstance(type, str):
```

Comment:
Same problem with Python 2 here, but in this instance it shouldn't matter, because type should be ASCII only.